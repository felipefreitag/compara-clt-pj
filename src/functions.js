function handleChange(event, employee) {
  const { value } = event.target;
  (employee ? this.setState({ employeeMonthly: value })
    : this.setState({ companyMonthly: value })
  )
}

function handleEmployeeChange(event) {
  this.handleChange(event, true)
}

function handleCompanyChange(event) {
  this.handleChange(event, false)
}

function inssCalc(salary) {
  let result = 0;
  if (salary <= 1659.38 ) {
    result= salary * 0.08
  } else if (salary <= 2765.66) {
    result= salary * 0.09
  } else if (salary <= 5531.31) {
    result= salary * 0.11
  } else result= (5531.31 * 0.11);
  return result;
}

function irrfCalc(salary) {
  let result = 0;
  if (salary <= 1903.98) {
    result = 0
  } else if (salary <= 2826.65) {
    result = (salary * 0.075 - 142.80)
  } else if (salary <= 3751.05) {
    result = (salary * 0.15 - 354.80)
  } else if (salary <= 4664.68) {
    result = (salary * 0.225 - 636.13)
  } else result = (salary * 0.275 - 869.36);
  return result;
}

function transport(salary) {
  const pass = 4.05
  const monthTransport = 4.05 * 2 * 22
  const percent = salary * 0.06
  return monthTransport > percent ? percent : monthTransport
}

function netSalary(salary) {
  const inss = inssCalc(salary);
  const irrf = irrfCalc(salary - inss);
  const netValue = salary - inss - irrf;
  return netValue;
}

function employeeYearlyEarnings(employeeMonthly) {
  const grossEarning = parseFloat(employeeMonthly);
  const netMonthly = netSalary(grossEarning);
  const vacations = netSalary(grossEarning * 1.3);
  // Total: 11 monthly payments + 1 with vacations + 13ยบ
  const yearTotal = 12 * netMonthly + vacations;
  return yearTotal.toLocaleString('de-DE', { maximumFractionDigits: '2'});
}

function employeeResults(employeeMonthly) {
  const grossEarning = parseFloat(employeeMonthly);
  const fgts = grossEarning * 13.3 * 0.08;
  return {
    name: "employee",
    yearly: employeeYearlyEarnings(employeeMonthly),
    fgts: fgts.toLocaleString('de-DE', { maximumFractionDigits: '2'}),
  }
}

function companyYearlyEarnings(companyMonthly) {
  const grossEarning = parseFloat(companyMonthly);
  const netMonthly = grossEarning - 60;
  const yearTotal = 12 * netMonthly;
  return yearTotal.toLocaleString('de-DE', { maximumFractionDigits: '2'});
}

function companyResults(companyMonthly) {
  return {
    name: "company",
    yearly: companyYearlyEarnings(companyMonthly),
  }
}

export { handleChange, handleEmployeeChange, handleCompanyChange }
export { employeeResults, companyResults }
export { inssCalc }
