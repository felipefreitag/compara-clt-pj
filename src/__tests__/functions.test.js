import * as subject from '../functions'

describe('inssCalc', () => {
  it('returns the correct value', () => {
    expect(subject.inssCalc(1000)).toBe(80)
    expect(subject.inssCalc(2000)).toBe(180)
    expect(subject.inssCalc(5000)).toBe(550)
    expect(subject.inssCalc(10000)).toBeCloseTo(608.44, 2)
  })
})

describe('irrfCalc', () => {
  it('returns the correct value', () => {
    expect(subject.irrfCalc(1000)).toBe(0)
    expect(subject.irrfCalc(2000)).toBeCloseTo(7.20, 2)
    expect(subject.irrfCalc(3000)).toBeCloseTo(95.2, 2)
    expect(subject.irrfCalc(4000)).toBeCloseTo(263.87, 2)
    expect(subject.irrfCalc(10000)).toBeCloseTo(1880.64, 2)
  })
})

describe('transport', () => {
  it('returns the correct value', () => {
    expect(subject.transport(1000)).toBe(60)
    expect(subject.transport(2000)).toBe(120)
    expect(subject.transport(5000)).toBe(178.2)
  })
})

describe('netSalary', () => {
  it('returns the correct value', () => {
    expect(subject.netSalary(1000)).toBe(920)
    expect(subject.netSalary(2000)).toBe(1820)
    expect(subject.netSalary(5000)).toBeCloseTo(4084.88, 0)
    expect(subject.netSalary(10000)).toBeCloseTo(7678, 0)
  })
})

describe('employeeYearlyEarnings', () => {
  it('returns the correct value', () => {
    expect(subject.employeeYearlyEarnings(1000)).toBe("12,236")
    expect(subject.employeeYearlyEarnings(2000)).toBe("24,171.35")
    expect(subject.employeeYearlyEarnings(5000)).toBe("54,159.3")
    expect(subject.employeeYearlyEarnings(10000)).toBe("101,992.09")
  })
})

describe('employeeResults', () => {
  it('returns the correct value', () => {
    [1000, 2000, 5000, 10000].map((value) => {
      expect(subject.employeeResults(value)).toMatchSnapshot()
    })
  })
})

describe('companyYearlyEarnings', () => {
  it('returns the correct value', () => {
    [1000, 2000, 5000, 10000].map((value) => {
      expect(subject.companyYearlyEarnings(value)).toMatchSnapshot()
    })
  })
})

describe('companyResults', () => {
  it('returns the correct value', () => {
    [1000, 2000, 5000, 10000].map((value) => {
      expect(subject.companyResults(value)).toMatchSnapshot()
    })
  })
})

describe('translate', () => {
  it('returns the translated strings', () => {
    expect(subject.translate('employee')).toEqual('CLT')
    expect(subject.translate('company')).toEqual('PJ')
    expect(subject.translate('yearly')).toEqual('Em um ano, vocÃª recebe:')
    expect(subject.translate('fgts')).toEqual('FGTS acumulado:')
    expect(subject.translate('foo')).toEqual('foo')
  })
})

describe('omit', () => {
  it('returns an empty object without argument', () => {
    expect(subject.omit()).toEqual({})
  })

  it('returns the object without the key to omit', () => {
    const obj = { foo: 'bar', baz: 12 }
    expect(subject.omit(obj, 'foo')).toEqual({ baz: 12 })
  })
})
