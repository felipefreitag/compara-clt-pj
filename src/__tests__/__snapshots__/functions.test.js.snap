// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`companyResults returns the correct value 1`] = `
Object {
  "name": "company",
  "yearly": "11,280",
}
`;

exports[`companyResults returns the correct value 2`] = `
Object {
  "name": "company",
  "yearly": "23,280",
}
`;

exports[`companyResults returns the correct value 3`] = `
Object {
  "name": "company",
  "yearly": "59,280",
}
`;

exports[`companyResults returns the correct value 4`] = `
Object {
  "name": "company",
  "yearly": "119,280",
}
`;

exports[`companyYearlyEarnings returns the correct value 1`] = `"11,280"`;

exports[`companyYearlyEarnings returns the correct value 2`] = `"23,280"`;

exports[`companyYearlyEarnings returns the correct value 3`] = `"59,280"`;

exports[`companyYearlyEarnings returns the correct value 4`] = `"119,280"`;

exports[`employeeResults returns the correct value 1`] = `
Object {
  "fgts": "1,064",
  "name": "employee",
  "yearly": "12,236",
}
`;

exports[`employeeResults returns the correct value 2`] = `
Object {
  "fgts": "2,128",
  "name": "employee",
  "yearly": "24,171.35",
}
`;

exports[`employeeResults returns the correct value 3`] = `
Object {
  "fgts": "5,320",
  "name": "employee",
  "yearly": "54,159.3",
}
`;

exports[`employeeResults returns the correct value 4`] = `
Object {
  "fgts": "10,640",
  "name": "employee",
  "yearly": "101,992.09",
}
`;
